LabTools/DataManagement.py

```diff
@@ -109,6 +111,13 @@ def update_user_variables(self, adict):
 
             self.user_variables = adict
 
+    def update_user_data(self, adict):
+        """
+        Replaces the user data with updated ones
+        """
+        if isinstance(adict, dict):
+            self.user_data = adict
```

Comment:
you can also use the `update` builtin function from the `dict`

Reply:
I added this to replace it entirely, to be used by LoadPlotWidget